### DISCUSSION TITLE
Dealing with bootstrap in html files

### DISCUSSION THREAD

**Discussion by m-weintraub** [jgm/pandoc#9497] [Q&A] [answered]
Hi.  I'm new to pandoc and got here via pypandoc. (and it looks like the google group is unavailable). I've got a question and am looking for some guidance.

I'm trying to take an html that uses bootstrap (5) and create pdf or docx files.  From what I've read, it looks like I should be defining a template.   But before I try to figure out what that means, I was hoping some good soul had already solved this.  So my questions:
1. How do you do this - apply the css on the input and have the output reflect something similar?   (and I'm not trying to solve this for all possible css)
2. Is there any existing work that I could leverage?

Thanks,
   Mike

Here's a better description:
I’ve got an html document that uses bootstrap 5 for formatting.  I can’t figure out how to get the css to carry through the process of translating an html doc to word or pdf.  I figured bootstrap, being a quasi-standard, fairly high-level specification, might be good way to structure the presentation. I chose to use bootstrap’s d-flex layout structure because I was trying to keep things fairly high level and I was only considering the browser and device independence.  But the client asked for html to be saved as pdf or docx, which is how I came to pandoc.

Mostly I’m trying to control what goes next to what, as in there is a question and there is an answer.   The desire is to put the question to the left of the answer.  In HTML, there’s not a good way to do this without some kind of augmentation.  I could use a table, and I could put the question into a column and the answer in a column (but that's a hack).  Or I could use a description list (but that's a hack).

Here's a specific example:
```
<div class=”d-flex flex-row”>
                <div>_Question_</div>
                <div>_Answer_</div>
</div>
 
```
In a browser, this gives (the outer div says to put the child div’s on the same line):
_Question Answer_
 
After pypandoc processing, it gives
 
Question
Answer
 
As in,
```
<div>
<div>
<div>
```
I can’t figure out what I need to do to incorporate the bootstrap classes so they pass through to the output.

And for completeness, here's how I'm  using pandoc:
```
pypandoc.convert_text(source=person_text, 
                      format='html',
                      to=target, 
                      outputfile=file_name,
                      extra_args=["-M2GB", "+RTS", "-K64m", "-RTS"])
```

**Comment by jgm** [comment, marked as answer]
Pandoc doesn't pay any attention to CSS, so this isn't a viable path if your target is docx.
If your target is pdf, you can simply use wkhtmltopdf or a similar tool to turn HTML directly into PDF.

**Reply by m-weintraub** [reply]
Alright.  I appreciate the counseling.
